texastribune/donations
# Add donation type fallback

#### What's this PR do?
Falls back to a donation type getter based off of a subscription's product when no donation type is provided. Donation type is provided automatically from all of our pages, so this only applies to donations created through the stripe dashboard.

#### Why are we doing this? How does it help us?
Currently stripe dashboard donations can't make it through to salesforce because they don't pass this info on through metadata like other donations do.

#### How should this be manually tested?
* In stripe, make sure you're in Test mode (upper right hand corner)
* Click the plus button (upper right hand corner) and choose "Subscription"
    * <img width="432" alt="Screen Shot 2023-10-13 at 2 47 20 PM" src="https://github.com/texastribune/donations/assets/88053677/4359975c-78c6-40d2-aaae-8006b1b8d756">
* Click in the "Customer" field and choose any of the options (likely me)
* Click in the "Product" field and choose any circle product (founders, editors, chairmans, leadership)
* Lastly, towards the bottom of the form, make sure a payment form is chosen
    * you'll see the last 4 digits (likely 4242)
* Click "Create Test Subscription" at the bottom of the form
* You should see the donation come through on #tech-test in slack

#### How should this change be communicated to end users?
I'll reach out to membership and Anna.

#### Are there any smells or added technical debt to note?
There are many possible solutions to this problem and I'd like to see if I could find a cleaner one once we update the front-end and can rethink our handling of product ids at that point.

Also worth considering if there is a place for admins to do this kind of donation add within ump at some point. Circle members in particular seem to pop up in this scenario, both because they ask admins to handle it, and because circle memberships aren't set it/forget it since they end after 3 years.

#### What are the relevant tickets?
